src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -116,17 +116,6 @@ private abstract class CompilationTrackerState
 
                 public CompilationTrackerGeneratorInfo GeneratorInfo { get; }
 
-                /// <summary>
-                /// Specifies whether <see cref="FinalCompilationWithGeneratedDocuments"/> and all compilations it depends on contain full information or not. This can return
-                /// <see langword="null"/> if the state isn't at the point where it would know, and it's necessary to transition to <see cref="FinalState"/> to figure that out.
-                /// </summary>
-                public virtual bool? HasSuccessfullyLoaded => null;
-
-                /// <summary>
-                /// The final compilation is potentially available, otherwise <see langword="null"/>.
-                /// </summary>
-                public virtual Compilation? FinalCompilationWithGeneratedDocuments => null;
```

Comment:
thse both become non-null in finalstate.  they are never present anywhere else.  i found it much clearer for people who were trying to read this to simply ask if they were in that state or not.